arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -831,6 +874,21 @@ Result DatabaseInitialSyncer::fetchCollectionDump(arangodb::LogicalCollection* c
       "&includeSystem=" + std::string(_config.applier._includeSystem ? "true" : "false") +
       "&useEnvelope=false" +
       "&serverId=" + _state.localServerIdString;
+  
+  if (ServerState::instance()->isDBServer() &&
+      !_config.isChild() &&
+      _config.applier._skipCreateDrop &&
+      _config.applier._restrictType == ReplicationApplierConfiguration::RestrictType::Include &&
+      _config.applier._restrictCollections.size() == 1 &&
+      !hasDocuments(*coll)) {
+    // DB server doing shard synchronization. now try to fetch everything in a single VPack array.
+    // note: only servers >= 3.10 will honor this URL parameter. servers that are not capable of
+    // this format will simply ignore it and send the old format.
+    // the syncer has code to tell the two formats apart.
+    // note: we can only add this flag if we are sure there are no documents present locally.
+    // everything else is not safe.
+    baseUrl += "&single=true";
```

Comment:
I think the name `single` is rather confusing in this context. We either get a "single array" or a bunch of "single vpack slices". Without the comment it is not really clear which one `single=true` is used for. Can we rename it to `array`?

Reply:
Renamed!